src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.Proxy.cs

```diff
@@ -322,21 +322,107 @@ public async Task Proxy_SslProxyUnsupported_Throws()
             }
         }
 
-        [OuterLoop("Uses external server")]
         [Fact]
-        public async Task Proxy_SendSecureRequestThruProxy_ConnectTunnelUsed()
+        public async Task ProxyTunnelRequest_GetAsync_Success()
         {
+            const string Content = "Hello world";
+
             using (LoopbackProxyServer proxyServer = LoopbackProxyServer.Create())
             {
                 HttpClientHandler handler = CreateHttpClientHandler();
                 handler.Proxy = new WebProxy(proxyServer.Uri);
+                handler.ServerCertificateCustomValidationCallback = TestHelper.AllowAllCertificates;
                 using (HttpClient client = CreateHttpClient(handler))
-                using (HttpResponseMessage response = await client.GetAsync(Configuration.Http.SecureRemoteEchoServer))
                 {
-                    Assert.Equal(HttpStatusCode.OK, response.StatusCode);
-                    _output.WriteLine($"Proxy request line: {proxyServer.Requests[0].RequestLine}");
-                    Assert.Contains("CONNECT", proxyServer.Requests[0].RequestLine);
+                    var options = new LoopbackServer.Options { UseSsl = true };
+                    await LoopbackServer.CreateServerAsync(async (server, uri) =>
+                    {
+                        Task<HttpResponseMessage> clientTask = client.GetAsync(uri);
+                        await server.AcceptConnectionSendResponseAndCloseAsync(content: Content);
+                        using (var response = await clientTask)
+                        {
+                            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                            Assert.Equal(Content, await response.Content.ReadAsStringAsync());
+                        }
+                    }, options);
                 }
+
+                Assert.Contains("CONNECT", proxyServer.Requests[0].RequestLine);
+            }
+        }
+
+        [Fact]
+        public async Task ProxyTunnelRequest_MaxConnectionsSetButDoesNotApplyToProxyConnect_Success()
```

Comment:
Should we set the pool idle timeout to infinite here? I assume the failure condition for this test relies on

https://github.com/dotnet/runtime/blob/90899319a8be56695a917c61d628de7e5b461991/src/libraries/Common/src/System/Net/Http/HttpHandlerDefaults.cs#L28

being larger than

https://github.com/dotnet/runtime/blob/90899319a8be56695a917c61d628de7e5b461991/src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs#L19

otherwise it could pass even without this change (after the first pool expired)

Reply:
We accept both connections before sending the responses. So it shouldn't matter.